### DISCUSSION TITLE
sea-orm-cli Support a flag for per-field model attributes

### DISCUSSION THREAD

**Discussion by andres-movl** [SeaQL/sea-orm#1799] [Ideas]
Started as a discussion on Discord

## Motivation

We are using [ActixAdmin](https://lib.rs/crates/actix-admin) which also uses SeaORM. However [ActixAdmin](https://lib.rs/crates/actix-admin) requires attributes on the model structs on a field level to specify which fields to show in the admin. The issue is that if we have any model changes, sea-orm-cli overrides everything so we need a new flag similar to `--model-extra-derives` but which allows the user to add attributes on a per field granularity. 

## Proposed Solutions

A solution would be to add a new flag to support "model-field-extra-derives". 

## Current Workarounds

None, just re-adding field level attributes manually every time there's a change to the models.

**Comment by BadMachine** [comment]
Any progress?

**Comment by Expurple** [comment]
> re-adding field level attributes manually every time there's a change to the models.

As a workaround for now, you can automate this step:

1. Use `diff` / `git diff` to save a `.patch` file with the diff that adds the attributes. Commit that file to the repo.
2. After generating models, invoke the `patch` utility to apply that diff and fix the file.

I recently had to do this because of this bug in model generation: #2409. I already had a wrapper script for code generation, to call `rustfmt` after `sea-orm-cli`.
